src/project.ts

```diff
@@ -220,8 +220,19 @@ export class AtlasDataset extends BaseAtlasClass {
    * @param ids A list of identifiers to fetch from the server.
    */
 
-  async fetch_ids(ids?: string[]): Promise<Record<string, any>[]> {
-    throw new Error('Not implemented');
+  async fetch_ids(
+    ids?: string[]
+  ): Promise<Record<string, Record<string, any>>> {
+    if (ids === undefined) {
+      return {};
+    }
+    const response = await this.apiCall(
+      '/v1/project/data/get',
+      'POST',
+      { project_id: this.id, datum_ids: ids },
+      null
+    );
+    return response as Record<string, Record<string, any>>;
```

Comment:
Do we need this as? I figured the Promise wrapper itself puts a type. The second assertion doesn't do anything

Reply:
We do need this to pass type checking, yes.